src/server/config_registry.cc

```
@@ -15,10 +16,27 @@ namespace {
 using namespace std;
 
 string NormalizeConfigName(string_view name) {
-  return absl::StrReplaceAll(name, {{"-", "_"}});
+  return absl::StrReplaceAll(name, {{"-", "_"}, {".", "_"}});
 }
 }  // namespace
 
+// Convert internal flag name back to user-facing format
+// Example: search_query_string_bytes -> search.query-string-bytes
+string DenormalizeConfigName(string_view name) {
```

why it is needed?

---

does the test require it?
I am fine that we silently accept `config set search.query-string-bytes` but display `search_query_string_bytes` as it is the way to set flags in dragonfly. the question whether the test breaks?

---

This is one of test's check (`test_query_string_bytes_limit`)

```
        assert client.execute_command("CONFIG GET search.query-string-bytes") == [b"search.query-string-bytes", b"10240"]
```